addons/crm/wizard/crm_lead_to_opportunity.py

```diff
@@ -82,6 +81,12 @@ def _compute_duplicated_lead_ids(self):
                 convert.lead_id.partner_id.email if convert.lead_id.partner_id.email else convert.lead_id.email_from,
                 include_lost=True).ids
 
+    @api.depends('partner_id')
+    def _compute_company_partner_id(self):
+        for convert in self.filtered('partner_id'):
+            if convert.company_partner_id and convert.partner_id.parent_id and not convert.company_partner_id.filtered_domain([('id', 'parent_of', convert.partner_id)]):
```

Comment:
If we change partner_id wo parent to one that has a parent, isn't company_partner_id updated to reflect it ? Aka: is the "if convert.company_partner_id" not a bit restrictive ?